### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by gviejo**
Adding `core_functions.py` to dispatch calls to numba core functions or pynajax

### CODE REVIEW

**Review by BalzaniEdoardo — CHANGES_REQUESTED** [review]
just a few comments! good job, that was a ton of stuff

File: docs/external.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Pynapple has been designed as a lightweight package for representing time series and epochs in system neuroscience.
+As such, it can function as a foundational element for other analysis packages handling time series data. Here we keep track of external projects that uses pynapple.
+
+
+## NEMOS
```
[the remark this answers is not in the document]
Comment (gviejo) [reply docs/external.md:5]: You are welcome

File: pynapple/core/_jitted_functions.py [hunk 2]
```diff
@@ -70,66 +73,108 @@ def jittsrestrict(time_array, starts, ends):
...
 
 
 @jit(nopython=True)
-def jitrestrict_with_count(time_array, data_array, starts, ends):
-    n = len(time_array)
-    m = len(starts)
-    ix = np.zeros(n, dtype=np.bool_)
-    count = np.zeros(m, dtype=np.int64)
+def jitvaluefrom(time_array, time_target_array, count, count_target, starts, ends):
+    m = starts.shape[0]
```
[the remark this answers is not in the document]
Comment (gviejo) [reply pynapple/core/_jitted_functions.py:89]: Someday

File: pynapple/core/time_series.py [hunk 3]
```diff
@@ -353,34 +440,30 @@ def dropna(self, update_time_support=True):
...
-                return self[~index_nan]
-
+                ep = None
         else:
-            return self
+            ep = self.time_support
+
+        kwargs = {}
+        if hasattr(self, "columns"):
+            kwargs["columns"] = self.columns
```
[the remark this answers is not in the document]
Comment (gviejo) [reply pynapple/core/time_series.py:456]: THanks

**Review by BalzaniEdoardo — APPROVED** [review]
great

**Pull request merged**